specs/governance/alligator.md

```diff
@@ -0,0 +1,207 @@
+  - [`AllowanceType`](#allowancetype)
+  - [`Checkpoint`](#checkpoint)
+- [Backwards Compatibility](#backwards-compatibility)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `Alligator` contract implements subdelegations for the [`GovernanceToken`](gov-token.md). Subdelegations enable
+advanced delegation use cases, such as partial, time-constrained & block-based delegations, and relative & fixed allowances.
+
+The `Alligator` contract migrates the delegation state from the [`GovernanceToken`](gov-token.md) to itself
+through a hook-based approach. Specifically, the [`GovernanceToken`](gov-token.md) calls the `Alligator` contract's
+`afterTokenTransfer` function after a token transfer. This enables the `Alligator` contract to consume the hook and update
+its delegation and checkpoint mappings accordingly. If either address involved in the transfer (`_from_` or `_to`) has not
+been migrated to the `Alligator` contract, the contract copies the address' checkpoint data from the
+[`GovernanceToken`](gov-token.md) to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `subdelegate`
+
+Allows subdelegation of token voting power to another address (delegatee) with specified subdelegation rules. This function
+is inteded to be called by users that require advanced delegation of the [`GovernanceToken`](gov-token.md).
+
+```solidity
+subdelegate(address _delegatee, SubdelegationRule _rule)
+```
+
+#### `subdelegateFromToken`
+
+Allows subdelegation of token voting power from an address (delegator) to another address (delegatee) with a specified
+subdelegation rule. This function is intended to be called by the [`GovernanceToken`](gov-token.md) contract as part
+of its `delegateBySig` function. To ensure backwards compatibility in the [`GovernanceToken`](gov-token.md), the
+subdelegation rule must be 100% delegation, mimicking the behavior of the [`GovernanceToken`](gov-token.md)'s `delegate`
+function.
+
+
+```solidity
+subdelegateFromToken(address _delegator, address _delegatee, SubdelegationRule _rule)
+```
+
+#### `subdelegateBatched`
+
+Allows batch subdelegation of token voting power to multiple addresses with specified subdelegation rules. This
+function is intended to be called by users.
+
+```solidity
+subdelegateBatched(address[] _delegatees, SubdelegationRule[] _rules)
+```
+
+Calls the `subdelegate` function for each pair of `_delegatees` address and subdelegation rule.
+
+#### `afterTokenTransfer`
+
+Updates the voting power of two addresses (`_from` and `_to`) after a token transfer. This function MUST
+be called by the `_afterTokenTransfer` function in the [`GovernanceToken`](gov-token.md).
+
+```solidity
+afterTokenTransfer(address _from, address _to, uint256 _amount)
```

Comment:
We should also be sure to add language about the risk of this reverting, ie it would prevent `GovToken` from being transferred

Reply:
added in `Security Considerations` section